### DISCUSSION TITLE
直播间右侧聊天区域宽度自定义

### DISCUSSION THREAD

**Discussion by u2shana** [the1812/Bilibili-Evolved#4216] [功能建议 / Ideas]
![P8CM(U8AY%BF}% ~KJFXFSB](https://github.com/the1812/Bilibili-Evolved/assets/14146754/3998b392-feae-4c64-9628-b1046b1c5819)

就算是主流的16:9和21:9显示器，开启聊天区域的情况下，

画面上下必定会有黑边，而且弹幕还会在黑边上显示，一半在里一半在外

所以是否可以自定义聊天区域宽度以达到直播有效区域完美填充呢

**Comment by the1812** [comment]
做出来试了下, 16:9 和 21:9 都不太好看啊... 你的显示器是什么比例?

16:9 会非常挤
<img height="400" src="https://github.com/the1812/Bilibili-Evolved/assets/26504152/a64b36a8-268d-40fb-8e6c-ada210ba0739">

21:9 会非常宽
<img height="400" src="https://github.com/the1812/Bilibili-Evolved/assets/26504152/773a0034-2889-487e-a295-abab4698605d">

**Reply by u2shana** [reply]
16:9不应该这么小的啊，1L的图就是1080P16:9显示器chrome窗口最大化的情况

**Reply by the1812** [reply]
我截个全屏的图给你看看
![Snipaste_2023-07-30_23-55-12](https://github.com/the1812/Bilibili-Evolved/assets/26504152/f4d69c5a-c1cb-462e-9ecb-d190cf0e7856)

**Reply by u2shana** [reply]
如果没有开启下方的windows任务栏，确实是这种效果。事实上比例还会受到浏览器上方的地址栏菜单栏等已占用空间的影响。
所以这个宽度能自定义是最好的，甚至还希望能像浏览器开发者工具面板一样，能够移动到上下左右侧

**Reply by the1812** [reply]
可以加个拖动条来自定义宽度

**Reply by the1812** [reply]
先把做完的部分发了, 自定义宽度之后再加

**Reply by u2shana** [reply]
希望还能有个移到左边的功能

**Reply by the1812** [reply]
移位置的应该会做到 #4179 的功能里

**Comment by lovelyCARDINAL** [comment]
黑边还可能是b站转码干的好事
有时候直播分辨率会是1920*1088，莫名其妙多了8

<img width="283" alt="image" src="https://github.com/the1812/Bilibili-Evolved/assets/69918945/c215c2e0-57d1-4791-b88c-601588aff103">

**Reply by yinzhefuyun** [reply]
能不能做一个显示直播视频码率解锁的功能，现在被B站去掉了，能恢复吗？然后B站视频限制缓存速度显示也是限制9900Kbps，不知道这些做限制干什么

**Reply by u2shana** [reply]
https://greasyfork.org/zh-CN/scripts/443043-%E6%81%A2%E5%A4%8Db%E7%AB%99%E7%9B%B4%E6%92%AD%E8%A7%86%E9%A2%91%E7%A0%81%E7%8E%87%E6%98%BE%E7%A4%BA
这有现成的脚本，不过我感觉现在数值好像不太准
